dwst/scripts/particles.js

```diff
@@ -20,7 +20,32 @@ const specialChars = [
   '\\',
 ];
 
-export class InvalidParticles extends Error { }
+export class InvalidParticles extends Error {
+  constructor(expression, expected, expectedStringLength, remainder) {
+    super();
+    this.expression = expression;
+    this.expected = expected;
+    // this.got = got;
+    this.remainder = remainder;
+    this.expectedStringLength = expectedStringLength;
+  }
+  getGot() {
+    return this.remainder.substring(0, this.expectedStringLength);
+  }
+  getErrorPosition() {
```

Comment:
You could do the same here `get errorPosition() {`

Reply:
done